include/openthread/nat64.h

```diff
@@ -88,6 +88,127 @@ typedef struct otIp4Cidr
     uint8_t      mLength;
 } otIp4Cidr;
 
+/**
+ * @struct otNat64Counters
```

Comment:
We can remove the `@struct` line.  Doxygen maps it to the next definition (which is the `struct` automatically). I think we need/use `@struct` if the definition does not immediately follow.